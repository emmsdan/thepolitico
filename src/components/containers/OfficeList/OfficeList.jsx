import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { getOfficeRequestHandler } from '../../../store/reducers/office';
import Loader from '../../presentationals/Loader/Loader';
import { Link } from 'react-router-dom';

class OfficeList extends Component {
  componentDidMount() {
    this.props.getOfficeRequestHandler('', 'offices');
  }

  render() {
    const offices = this.props.offices.offices.map((office, key) => {
      return (
        <div
          style={{
            background: '#fff',
            padding: '10px',
            width: '300px',
            margin: '10px',
            textAlign: 'center',
            boxShadow: '0px 2px 12px #000',
          }}
          key={key}
        >
          <h2> {office.type} </h2>
          <div className="offices_office" style={{ height: 'auto' }}>
            <h3>
              <Link to={`/office/${office.id}/candidate`}> {office.name} </Link>
            </h3>
            <div className="offices_office_options">
              <Link to={`/office/${office.id}/candidate`}>
                <span style={{ color: '#fff' }}>view candidate</span>
              </Link>
            </div>
          </div>
        </div>
      );
    });

    return (
      <>
        {this.props.offices.isLoading && (
          <Loader loader="show" text="Hi, am processing your input" />
        )}
        <div className="offices">
          <br /> <br />
          {offices}
        </div>
      </>
    );
  }
}

OfficeList.propTypes = {
  getOfficeRequestHandler: PropTypes.func.isRequired,
  offices: PropTypes.object,
  match: PropTypes.object,
  isLoading: PropTypes.bool,
};

const mapStateToProps = state => {
  return { offices: state.offices };
};

export default connect(
  mapStateToProps,
  { getOfficeRequestHandler },
)(OfficeList);
