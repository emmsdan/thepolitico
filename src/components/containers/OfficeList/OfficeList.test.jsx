import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import OfficeList from './OfficeList';

const props = {
  getPartyRequestHandler: jest.fn(),
  isLoading: false,
  offices: [{}, {}],
};

let store;
describe('OfficeList', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({ ...props });
  });

  it('should render correctly', () => {
    const component = shallow(<OfficeList store={store} />);
    expect(component).toMatchSnapshot();
  });
});
