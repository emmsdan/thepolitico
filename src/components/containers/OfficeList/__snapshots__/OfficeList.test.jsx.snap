// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfficeList should render correctly 1`] = `
<ContextProvider
  value={null}
>
  <OfficeList
    getOfficeRequestHandler={[Function]}
    offices={
      Array [
        Object {},
        Object {},
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
